main.py

```diff
@@ -114,17 +114,14 @@ def _restore_links(self):
             endpoint_a.nni = True
             endpoint_b.nni = True
 
-            self.notify_topology_update()
+            if state['enabled']:
+                link.enable()
+            else:
+                link.disable()
 
-            try:
-                if state['enabled']:
-                    self.links[link_id].enable()
-                else:
-                    self.links[link_id].disable()
```

Comment:
Please check if by removing this, the instance of `link` that you're updating is really updating the global `link` instance that will be used by others.

Reply:
Thanks, I checked, the updated `link` instance is global.